core/src/avm1/globals/object.rs

```diff
@@ -178,22 +177,12 @@ pub fn register_class<'gc>(
 
     let class_name = class_name.coerce_to_string(activation)?;
 
-    let registry = activation
-        .base_clip()
-        .movie()
-        .map(|movie| activation.context.library.library_for_movie_mut(movie))
-        .and_then(|library| library.avm1_constructor_registry());
-
-    match registry {
-        Some(registry) => {
-            registry.set(class_name, constructor, activation.context.gc_context);
-            Ok(true.into())
-        }
-        None => {
-            log::warn!("Can't register_class without a constructor registry");
-            Ok(false.into())
-        }
-    }
+    activation.context.library.register_avm1_constructor(
+        activation.swf_version(),
```

Comment:
I'm not sure whether this preserves behavior in 100%. I assume that `activation.base_clip().movie().version()` is equivalent to `activation.swf_version()`.

Reply:
Not sure either but I think that `activation.swf_version()` always returns the version of the root SWF, while the other one can return the version of the inner SWFs (if there are any - otherwise it returns the version of the root SWF).

Reply:
I should point out that other parts of this PR (in display objects) use `self.movie().version()`, which can differ from `activation.swf_version()`. Furthermore, the SWF version is pulled specifically to select if we're using the case-sensitive or case-*in*sensitive registry; which have different contents. So if you loaded a SWF6 movie into a SWF7 root, then `Object.registerClass` would be registering case-sensitive classes. This might already be bad enough, but `MovieClip` would be looking for case-insensitive classes, and as far as I can tell there's no overlap between the registries. If a SWF6 movie registers a class, SWF7 doesn't see it. And with this code using the root version, loaded SWF6 movies can't actually see their classes either.

Alternatively, it could be the case that Flash uses the root clip's version in this case; in which case this code is fine and the `MovieClip` code further down in this PR needs to be changed. We probably should have a test case for this...

Reply:
Seems like `activation.swf_version()` is wrong, as it returns the version of the root/parent when a child calls to `Object.registerClass()`. So I switched back to `activation.base_clip().movie().version()` (this time with an `unwrap()`), and added tests to cover all 4 combinations of parent/child versions.